darts/models/forecasting/forecasting_model.py

```diff
@@ -1226,49 +1226,54 @@ def gridsearch(
-                    series=series,
-                    past_covariates=past_covariates,
-                    future_covariates=future_covariates,
-                    num_samples=1,
-                    start=start,
-                    forecast_horizon=forecast_horizon,
-                    stride=stride,
-                    metric=metric,
-                    reduction=reduction,
-                    last_points_only=last_points_only,
-                    verbose=verbose,
-                )
-            else:  # split mode
-                model._fit_wrapper(series, past_covariates, future_covariates)
-                pred = model._predict_wrapper(
-                    len(val_series),
-                    series,
-                    past_covariates,
-                    future_covariates,
-                    num_samples=1,
-                    verbose=verbose,
-                )
-                error = metric(val_series, pred)
+                if param_combination_dict.get("model_name", None):
+                    current_time = time.strftime("%Y-%m-%d_%H.%M.%S.%f", time.localtime())
+                    param_combination_dict[
+                        "model_name"
+                    ] = f"{current_time}_{param_combination_dict['model_name']}"
+
+                model = model_class(**param_combination_dict)
+                if use_fitted_values:  # fitted value mode
+                    model._fit_wrapper(series, past_covariates, future_covariates)
+                    fitted_values = TimeSeries.from_times_and_values(
+                        series.time_index, model.fitted_values
+                    )
+                    error = metric(series, fitted_values)
+                elif val_series is None:  # expanding window mode
+                    error = model.backtest(
+                        series=series,
+                        past_covariates=past_covariates,
+                        future_covariates=future_covariates,
+                        num_samples=1,
+                        start=start,
+                        forecast_horizon=forecast_horizon,
+                        stride=stride,
+                        metric=metric,
+                        reduction=reduction,
+                        last_points_only=last_points_only,
+                        verbose=verbose,
+                    )
+                else:  # split mode
+                    model._fit_wrapper(series, past_covariates, future_covariates)
+                    pred = model._predict_wrapper(
+                        len(val_series),
+                        series,
+                        past_covariates,
+                        future_covariates,
+                        num_samples=1,
+                        verbose=verbose,
+                    )
+                    error = metric(val_series, pred)
 
-            return float(error)
+                return float(error)
+            except Exception as e:
```

Comment:
I think it's a good idea, however this `try` block is quite huge and it'd be good to narrow it down if possible. I assume the errors we're trying to catch would happen in fit/predict calls, right? In that case, could we only have try/catch block(s) around the relevant calls?